test/end-to-end/company.js

```diff
@@ -227,6 +228,229 @@ describe("Company endpoint", () => {
+                            owner: test_company_3._id,
+                            ownerName: test_company_3.name,
+                            ownerLogo: test_company_3.logo
+                        })
+                    )).sort((a1, a2) => a1.publishDate > a2.publishDate));
+
+                const res = await test_agent
+                    .get(`/company/${test_company_3._id}/profile`)
+                    .expect(HTTPStatus.OK);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers.length).toEqual(CompanyConstants.offers.max_profile_visible);
+                for (let idx = 0; idx < CompanyConstants.offers.max_concurrent; idx++) {
+                    const val = offers[idx];
+
+                    expect(res.body.offers[idx].owner.toString()).toEqual(val.owner.toString());
+                    expect(res.body.offers[idx].ownerName.toString()).toEqual(val.ownerName.toString());
+                    expect(res.body.offers[idx].ownerLogo.toString()).toEqual(val.ownerLogo.toString());
+                }
+            });
+
+            test("should return if a random user is unable to see hidden offers", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user)
+                    .expect(HTTPStatus.OK);
+
+                await Offer.create(
+                    generateTestOffer({
+                        isHidden: true,
+                        owner: hidden_company._id.toString(),
+                        ownerName: hidden_company.name,
+                        ownerLogo: hidden_company.logo,
+                    })
+                );
+
+                const res = await test_agent
+                    .get(`/company/${hidden_company._id}/profile`)
+                    .expect(HTTPStatus.OK);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers.length).toEqual(0);
+            });
+
+            test("should return if a random user is unable to see disabled offers", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user)
+                    .expect(HTTPStatus.OK);
+
+                const disabled_test_offer = await Offer.create(
+                    generateTestOffer({
+                        owner: disabled_company._id.toString(),
+                        ownerName: disabled_company.name,
+                        ownerLogo: disabled_company.logo,
+                    })
+                );
+
+                await (new OfferService()).disable(disabled_test_offer._id, OfferConstants.HiddenOfferReasons.ADMIN_BLOCK);
+
+                const res = await test_agent
+                    .get(`/company/${disabled_company._id}/profile`)
+                    .expect(HTTPStatus.OK);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers.length).toEqual(0);
```

Comment:
This name is missleading. The company is not disabled, one of its offers is. Also, what happens if we try to access a disabled or blocked company's page?